tests/Feature/Events/ExportTest.php

```diff
@@ -76,12 +76,58 @@
 
         // Export devices.
         $response = $this->get("/export/devices");
+        $header = $response->headers->get('content-disposition');
+        $filename = public_path() . '/' . substr($header, strpos($header, 'filename=') + 9);
 
-        $filename = 'public/devices.csv';
         $fh = fopen($filename, 'r');
         fgetcsv($fh);
         $row2 = fgetcsv($fh);
-        self::assertEquals(e($event1->getEventName()), e($row2[6]));
+        self::assertEquals(e($event1->getEventName()), e($row2[7]));
+        $row3 = fgetcsv($fh);
+        self::assertEquals(e($event2->getEventName()), e($row3[7]));
+
+        // Export devices for a particular event.
+        $response = $this->get("/export/devices/event/$idevents1");
+        $header = $response->headers->get('content-disposition');
+        $filename = public_path() . '/' . substr($header, strpos($header, 'filename=') + 9);
+        $fh = fopen($filename, 'r');
```

Comment:
@edwh Commenting in case you haven't seen these - quite handy to see GH/Sonar inline these checks in to code reviews now.

That said, I figure this is not an issue in this case, give this is in a test.